spring-boot-starter/starter-client/spring/src/main/java/org/camunda/bpm/client/spring/impl/subscription/SubscriptionPostProcessor.java

```diff
@@ -0,0 +1,106 @@
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.camunda.bpm.client.spring.impl.subscription;
+
+import org.camunda.bpm.client.spring.impl.client.util.ClientLoggerUtil;
+import org.camunda.bpm.client.spring.annotation.ExternalTaskSubscription;
+import org.camunda.bpm.client.spring.impl.subscription.util.SubscriptionLoggerUtil;
+import org.camunda.bpm.client.spring.impl.util.AnnotationUtil;
+import org.camunda.bpm.client.task.ExternalTaskHandler;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.ListableBeanFactory;
+import org.springframework.beans.factory.annotation.AnnotatedBeanDefinition;
+import org.springframework.beans.factory.config.BeanDefinition;
+import org.springframework.beans.factory.config.ConfigurableListableBeanFactory;
+import org.springframework.beans.factory.support.BeanDefinitionBuilder;
+import org.springframework.beans.factory.support.BeanDefinitionRegistry;
+import org.springframework.beans.factory.support.BeanDefinitionRegistryPostProcessor;
+import org.springframework.core.type.AnnotatedTypeMetadata;
+
+public class SubscriptionPostProcessor implements BeanDefinitionRegistryPostProcessor {
+
+  protected static final SubscriptionLoggerUtil LOG = ClientLoggerUtil.SUBSCRIPTION_LOGGER;
+
+  protected Class<? extends SpringTopicSubscriptionImpl> springTopicSubscription;
+
+  public SubscriptionPostProcessor() {
+    this(SpringTopicSubscriptionImpl.class);
+  }
+
+  public SubscriptionPostProcessor(Class<? extends SpringTopicSubscriptionImpl> springTopicSubscription) {
+    this.springTopicSubscription = springTopicSubscription;
+  }
+
+  @Override
+  public void postProcessBeanDefinitionRegistry(BeanDefinitionRegistry registry) throws BeansException {
+    if (registry instanceof ListableBeanFactory) {
+      ListableBeanFactory listableBeanFactory = (ListableBeanFactory) registry;
+      String[] handlerBeans = listableBeanFactory.getBeanNamesForType(ExternalTaskHandler.class);
+      LOG.handlerBeansFound(ExternalTaskHandler.class, handlerBeans);
+
+      for (String handlerBeanName : handlerBeans) {
+        BeanDefinition beanDefinition = registry.getBeanDefinition(handlerBeanName);
+        ExternalTaskSubscription subscriptionAnnotation = findSubscriptionAnnotation(beanDefinition);
+        if (subscriptionAnnotation != null) {
+          SubscriptionConfiguration subscriptionConfiguration = new SubscriptionConfiguration();
+          subscriptionConfiguration.fromAnnotation(subscriptionAnnotation);
+          BeanDefinition subscriptionBeanDefinition = getBeanDefinition(handlerBeanName, subscriptionConfiguration);
+
+          String subscriptionBeanName = handlerBeanName + "Subscription";
```

Comment:
Alternative proposal:
* Let's use the topic name as the name for the subscription bean name so users can easily `@Autowired` those in their applications
* Let's add a defined suffix to the handler bean name and re-register the handler bean with that name, i.e. remove the old one and register it under the new name
* I would assume people do auto-wire the subscription bean more often to interact with it instead of the handler bean, so let's make it easier for them 😃

Reply:
👍

Reply:
I agree with the topic name as the bean name for subscriptions. However, I would propose to keep the handler naming as is. Here is why:
* A user defines the handler bean explicitly via {{@Bean}} annotation, so the user can fully control the handler bean name on its own
* When having spring support in your IDE (like in IntelliJ), you can navigate to the bean definition by clicking on `@Qualifier("handler-name")` which does not work if we add a suffix programmatically

Reply:
As discussed. Let's keep everything as it is. I will create a follow-up ticket where I document our discussion.

Reply:
I documented our discussion points and the decision in CAM-13255.